pkg/ipam/multipool.go

```diff
@@ -495,6 +499,10 @@ func (m *multiPoolManager) updateCiliumNode(ctx context.Context) error {
 		if err != nil {
 			return fmt.Errorf("failed to update node spec: %w", err)
 		}
+		// Update the pools in local node store.
+		m.localNodeStore.Update(func(n *node.LocalNode) {
+			n.IPAMAllocPools = newNode.AllocatedIPAMPools()
+		})
```

Comment:
> someone catch that event and update LocalNodeStore

That is not happening (verified by setting logging throughout). This is needed since https://github.com/cilium/cilium/pull/27285 updated BGP CP to use the LocalNodeStore.